zlibWrapper/zstd_zlibwrapper.c

```diff
@@ -67,29 +69,52 @@ static int g_useZSTD = ZWRAP_USE_ZSTD;   /* 0 = don't use ZSTD */
 
 void useZSTD(int turn_on) { g_useZSTD = turn_on; }
 
-int isUsingZSTD() { return g_useZSTD; }
+int isUsingZSTD(void) { return g_useZSTD; }
 
-const char * zstdVersion() { return ZSTD_VERSION_STRING; }
+const char * zstdVersion(void) { return ZSTD_VERSION_STRING; }
 
 ZEXTERN const char * ZEXPORT z_zlibVersion OF((void)) { return zlibVersion();  }
 
 
+static void* ZWRAP_allocFunction(void* opaque, size_t size)
+{
+    z_streamp strm = (z_streamp) opaque;
+    void* address = strm->zalloc(strm->opaque, 1, size);
+  /*  printf("ZWRAP alloc %p, %d \n", address, (int)size); */
+    return address;
+}
+
+static void ZWRAP_freeFunction(void* opaque, void* address)
+{
+    z_streamp strm = (z_streamp) opaque;
+    strm->zfree(strm->opaque, address);
+   /* if (address) printf("ZWRAP free %p \n", address); */
+}
+
+
 
 /* *** Compression *** */
 
 typedef struct {
     ZBUFF_CCtx* zbc;
     size_t bytesLeft;
     int compressionLevel;
+    z_stream allocFunc; /* copy of zalloc, zfree, opaque */
```

Comment:
I think I fixed all problems. This takes 10 commits. When I will publish them they will be in tihs (PR #198). Do you prefer them in a separate PR?